lib/diff.ml

```diff
@@ -130,32 +150,133 @@ let rec type_item ~typing_env ~name ~reference ~current =
                 modified_record_type ~typing_env ~ref_label_lst ~cur_label_lst
               in
               Some
-                (Type { tname = name; tdiff = Modified (Compound changed_lbls) })
-          | _, _ ->
+                (Type
+                   { tname = name; tdiff = Modified (Record_diff changed_lbls) })
+          | ( Type_variant (ref_constructor_lst, _),
+              Type_variant (cur_constructor_lst, _) ) ->
+              let changed_constrs =
+                modified_variant_type ~typing_env ~ref_constructor_lst
+                  ~cur_constructor_lst
+              in
+              Some
+                (Type
+                   {
+                     tname = name;
+                     tdiff = Modified (Variant_diff changed_constrs);
+                   })
```

Comment:
This is not entirely guaranteed to contain a diff. For instance, one could change the variance of one the type parameters without changing the concrete definition. There would be a diff for the type declaration but the individual contructors's definition would remain identical.

I think this also holds for the record case above.

Could you add a test that exhibits that, make sure that the current implementation fails said test and fix it?